### ISSUE TITLE
GTF file error

### ISSUE THREAD

**Issue by rsg344** [Townsend-Lab-Yale/cancereffectsizeR#14]
I've been trying to modify my gtf file in R so that I can create reference data. I've tried many different versions that I've pulled from the NCBI, USCS, and ensembl. However, I keep getting the error message: 

Column 3 of j's result for the first group is NULL. We rely on the column types of the first result to decide the type expected for the remaining groups (and require consistency). NULL columns are acceptable for later groups (and those are replaced with NA of appropriate type and recycled) but not for the first. Please use a typed empty vector instead, such as integer() or numeric().

I've tried modifying the gtf files on my own in R, but even if I try to replace the NULL with an empty vector such as integer() and then convert it back to a gtf file I can use, I keep getting the same issue.

Thank you so much!

**Comment by jeff-mandell** [comment]
I've released v2.6.2 of the package. If you re-install, then you should be able to build a canFam3 refset. The following worked for me:

```
library(data.table)
library(cancereffectsizeR) # 2.6.2 or later

# GTF source:
# http://ftp.ensembl.org/pub/release-104/gtf/canis_lupus_familiaris/Canis_lupus_familiaris.CanFam3.1.104.gtf.gz

# Load GTF as a data.table, and add chr prefixes to match BSgenome.Cfamiliaris.UCSC.canFam3
gtf = as.data.table(rtracklayer::import("Canis_lupus_familiaris.CanFam3.1.104.gtf.gz"))

# Chromosome names (seqnames) are 1, 2, ..., 38, X in the GTF
# Restrict to primary contigs and add chr prefixes for compatibility with canFam3 BSgenome
gtf = gtf[seqnames %in% c(1:38, 'X')]
gtf[, seqnames := paste0('chr', seqnames)]

# some genes have no names; let gene_id be the name for these
gtf[is.na(gene_name), gene_name := gene_id]

# NCBI-style names causes issues for this genome, so use UCSC-style.
refcds = build_RefCDS(gtf = gtf, genome = 'canFam3', chromosome_style = 'UCSC')

# Use exon regions to define the canine exome
exome = makeGRangesFromDataFrame(gtf[type == 'exon'])

# define supported chromosomes
supported_chr = paste0('chr', c(1:38, 'X'))

dir.create('canFam3_refset')
create_refset(output_dir = 'canFam3_refset', refcds_output = refcds, species_name = 'dog',
              genome_build_name = 'canFam3', BSgenome_name = 'canFam3', 
              supported_chr = supported_chr, default_exome = exome)
              
```

Other GTFs should work, too, as long as they have all the required annotations. Chromosome names might need to be adjusted. Let us know how it goes!

**Comment by rsg344** [comment]
Thank you so much for all of your help! That code works perfectly.
